sys/fido2/ctap/ctap_mem.c

```diff
@@ -136,9 +136,9 @@ static bool _flash_is_erased(uint32_t addr, size_t len)
 #endif
 }
 
-static uint32_t _flash_start_addr(void)
+static uintptr_t _flash_start_addr(void)
```

Comment:
this function is used in a number of places that need a bigger uint too

Reply:
i checked the return of this seems to be only used a 32bit addr-pointing into the backing memory 

i suggest return uint32_t
 (and document that the fill backing allways starts at 0 and will never be mapped to a region > 32Bit max)

and native  `_backing_memory` should be a unit32_t too ( to avoid confusion with a pointer into memory)

why do you need periph_flashpage_in_address_space if you are accessing everything through `mtd_*` functions

Reply:
The `periph_flashpage_in_address_space` is used to reserve flashpages such that different modules don't corrupt each others data by accidentally writing to each others flash pages. To my understanding, this unrelated to do with `mtd_*` functions.

Reply:
periph_flashpage_in_address_space says that that the flash is memory mapped and can thous be accessed through pointer dereferencing or memcpy strcpy ...  if every access to flash is trough mtd_* periph_flashpage_in_address_space is not need.

But i see you are unsing it for the `FLASH_WRITABLE_INIT` which has the feature add off linktime flash memory allocation.

Reply:
I adjusted the function to return a `uint32_t` again like it was originally. Any other change I should do ?

Reply:
not for this PR -- it is OK

please leave a comment reassuring me that you test ran this at the bottom.

Reply:
Will do a final test today evening and then comment